### DISCUSSION TITLE
3.3

### DISCUSSION THREAD

**Discussion by ccbogel** [ccbogel/QualCoder#796] [Announcements]
# QualCoder 3.3 Release

Binaries (double-click to run) are available at the end of the release page for:
Windows 10/11 QualCoder-3.3.exe    md5 checksum: 0b8f50b6d6c5e22d3cb17d2c78f2facb
Ubuntu 22.04 QualCoder-3.3-Ubuntu-22.04    md5 checksum: b7eff6944e0da74c444b710f451eae15

## Main
Import a codes list from a plain text file. One code name per line. No categories are applied.

## Accessibility
Throughout - added more key shortcuts. Under Help there is an option to display all the key shortcuts.
The colour selector shows how code colours can be seen with different types of colour impaired vision.
Under Coding menu there is a new window called Colour scheme. This allows a range of selected colours to be applied to a range of codes. This area also shows how the colours may be seen by people with different types of colour impairment.

## Manage files
Can hide columns by right-click on the table header row for options to hide that column or to hide columns beginning with specific text. Or to show columns where  the column name starts with specific text.
Can assign a file to a case via right-click menu when in the cases column.
Can right-click on a URL in an attribute and open the URL with the default web browser.
Faster table filling.

### Twitter data
Import twitter data from a fully quoted csv file.
Example file in Examples folder.
Requires these exact column names: _id_ and _full_text_ headers for tweet data and _screen_name_ for user data.
Additional tweet fields can be these exact column names: 
_created_at_, _coordinates_, _retweet_count_, _favorite_count_, _lang_
Additional user fields can be:
_location, url, description, followers_count, friends_count, listed_count, favourites_count, statuses_count_

The twitter import page also have an example of how to use R with RTweet to extract tweet data. This was tested and worked before twitter applied it fee for service. Note that currently there is a cost associated with accessing twitter data. I cannot give advice on any problems you may have getting twitter data.

The tweet data is loaded into individual database files (Manage Files), The user data is loaded into cases (Manage cases). Multiple tweets (stored as files are assigned to the matching user (case).

## Manage cases
Faster table filling
Can hide columns by right-click on the table header row for options to hide that column or to hide columns beginning with specific text. Or to show columns starting with a text value.
Can order by case name or by attribute value, ascending or descending.
Can hide rows by right-click menu and selecting a text value that equals or 'is like' some chosen text.

## Case file manager
Some improved functions in this dialog include a check box to hide assigned files. The right hand side text edit will fill with any file text when the file is selected in the left hand side table.

## Journals
Some markdown is enacted to make viewing journal entries nicer: 
\#, \#\#, \#\#\# for heading levels
**text** for bolded text
*text* for italic text

## Code text

'in vivo' coding added. Selected text becomes the code

## Colour scheme
There is a new window for setting colour ranges to codes. It also has options to see the codes from difference colour vision perspectives.

<hr /><em>This discussion was created from the release <a href='https://github.com/ccbogel/QualCoder/releases/tag/'>3.3</a>.</em>

**Comment by enricllagostera** [comment]
Thanks for this release. The tool is very usable and powerful. Just wanted to show some appreciation and say that I look forward to using it!

**Reply by ccbogel** [reply]
Thank you, I appreciate the positive feedback!
